### PR TITLE
Grupp b2019 w21#6058

### LINKED ISSUE #6058: Sorting doesn't work in resulted.php
In resulted.php sorting doesn't seem to work. Nothing happens when you press the sort button.

resultsorting

### PR DESCRIPTION
**Pull request by a17oscno**
Fixes #6058 

The filtering dropdown menu should now work in resulted with the 3 different ways that @HGustavs  wanted it to be able to sort by (acending, decending and pending).

Link: https://a17oscno.gruppb.webug.his.se:20002/DuggaSys/resulted.php?cid=2&coursevers=97732

### CODE REVIEW

File: DuggaSys/resulted.js [hunk 1]
```diff
@@ -317,12 +300,20 @@ function leaves() {
 
 	if (!(ocol == col && odir == dir) || typechanged) {
 		typechanged = false;
-		resort();
+    // This one is only here due to a bug where sometimes you need to sort multiple times to get the correct one.
```
Comment (JohanHaggmark) [inline DuggaSys/resulted.js]: The sorting logic in SortableTable is wrong. Maybe the bug is casued by that.

File: Shared/SortableTableLibrary/sortableTable.js [hunk 2]
```diff
@@ -778,10 +779,22 @@ function newCompare(firstCell, secoundCell) {
...
+			//Ascending grade
 			val = firstCellTemp.toLocaleUpperCase().localeCompare(secoundCellTemp.toLocaleUpperCase(), "sv");
+
+		} else if (status == 1) {
+			//return secoundCellTemp > firstCellTemp ? 1 : -1;
+			if(secoundCellTemp != "" && firstCellTemp != "" && secoundCellTemp != 0 && firstCellTemp != 0){
+				//descending grades
+				val = secoundCellTemp.toLocaleUpperCase().localeCompare(firstCellTemp.toLocaleUpperCase(), "sv");
+			}	
+		} else if (status == 2) {
```
Comment (JohanHaggmark) [inline Shared/SortableTableLibrary/sortableTable.js]: This should not be merged. A newer version exists in GruppB2019W21

Comment (c17johha) [inline Shared/SortableTableLibrary/sortableTable.js]: This should not be merged. A newer version exists in GruppB2019W21

File: DuggaSys/resulted.js (hunk 1 above)
Comment (c17johha) [inline DuggaSys/resulted.js]: SortableTable sorting logic should not be implemented. Maybe the updated sorting logic will eleminate this bug. the new logic can be found in GruppB2019W21

File: DuggaSys/resulted.js [hunk 3]
```diff
@@ -259,6 +240,7 @@ function makeCustomFilter(filtername, labeltext) {
 	}
 	if (filterList[filtername]) {
 		str += " checked";
+
```
Comment (c17johha) [inline DuggaSys/resulted.js]:
```suggestion
```

**Pull request merged**